api/pkg/gateway/context_test.go

```diff
@@ -0,0 +1,477 @@
+package gateway
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/labstack/echo/v4"
+
+	"github.com/shellhub-io/shellhub/pkg/api/authorizer"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRole(t *testing.T) {
+	cases := []struct {
+		description string
+		headers     map[string]string
+		expected    authorizer.Role
+	}{
+		{
+			description: "verify if given value returns from header",
+			headers: map[string]string{
+				"X-Role": authorizer.RoleOwner.String(),
+			},
+			expected: authorizer.Role("owner"),
+		}, {
+			description: "validate empty ID string, for ID behaviour",
+			headers: map[string]string{
+				"X-Role": "",
+			},
+			expected: authorizer.RoleInvalid,
+		}, {
+			description: "validate empty headers, for fail function",
+			headers:     map[string]string{},
+			expected:    authorizer.RoleInvalid,
+		},
```

Comment:
These two cases are redundant. `authorizer.RoleFromString` checks whether `X-Role` corresponds to a valid role. If it doesn't, it returns `authorizer.RoleInvalid`.

Since `echo` assigns an empty string to a non-existent header, both scenarios will produce the same result. If you prefer to maintain distinct cases, you can handle one case without the header (similar to the second one), and another with an invalid role (such as the string `invalid`).

Reply:
Understood